File: consorcio/lib/domain/value_objects/theme_preference.dart
class ThemePreference {
  static const String light = 'light';
  static const String dark = 'dark';

  static String normalize(String? value) {
    if (value == dark) return dark;
    return light;
  }

  static bool isDark(String? value) => normalize(value) == dark;

  static String toggle(String? value) =>
      isDark(value) ? light : dark;
}

File: src/domain/value-objects/ThemePreference.ts
export const ThemePreference = {
  Light: 'light',
  Dark: 'dark',
} as const

export type ThemePreference =
  (typeof ThemePreference)[keyof typeof ThemePreference]

export function isThemePreference(value: unknown): value is ThemePreference {
  return value === ThemePreference.Light || value === ThemePreference.Dark
}

export function normalizeThemePreference(value: unknown): ThemePreference {
  return isThemePreference(value) ? value : ThemePreference.Light
}
